bigtop-deploy/puppet/modules/hadoop/manifests/init.pp

```diff
@@ -365,6 +365,7 @@
   class common_mapred_app (
       $mapreduce_cluster_acls_enabled = undef,
       $mapreduce_jobhistory_host = undef,
+      $mapreduce_job_tracker_port = "8021",
```

Comment:
I thought job_tracker is MR1, which is no longer supported in Bigtop. No?

Reply:
ditto. 
Otherwise, the smoke tests will failed by:
1. giraph is running in local mode.
2. The test process will hang.

Reply:
mapred.job.tracker is only available in MRv1. See:
https://hadoop.apache.org/docs/r2.8.4/hadoop-mapreduce-client/hadoop-mapreduce-client-core/mapred-default.xml

Further more it seems that just having mapred.job.tracker specified to let Giraph pass env check is enough.
https://github.com/apache/giraph/blob/0c1e2ebce6e4f6e1a47d4dec5c0b8475433ef8df/giraph-core/src/main/java/org/apache/giraph/job/GiraphJob.java#L155

Hence you can set any value to mapred.job.tracker as long as it is not local. I've tested this.
Meanwhile, it seems that having mapreduce.jobtracker.address set to non local can also make tests passed as well. I guess somehow hadoop mapped mapreduce.jobtracker.address to mapred.job.tracker for backward compatibility.

Now, getting to my conclusion. I think what's best for this patch:
* Set mapred.job.tracker  with value = "BIGTOP-3137: Required by Giraph checkLocalJobRunnerConfiguration". Assuming that this value only effects Giraph
* Remove mapred.map.tasks
* Remove $mapreduce_job_tracker_port = "8021",

Reply:
It makes sense! I'll update it late. 
Really thanks for your awesome comments!

Reply:
@guyuqi Any update on this ;)

Reply:
Patch is updated and I've also tested it on ubuntu16.04.